lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -291,10 +291,21 @@ def headers
       end
 
       def success_from(response, action = nil)
-        if action == 'refund'
-          response.dig('transaction', 'status_detail') == 7 || SUCCESS_STATUS.include?(response.dig('transaction', 'current_status') || response['status'])
+        transaction_current_status = response.dig('transaction', 'current_status')
+        request_status = response['status']
+        transaction_status = response.dig('transaction', 'status')
+        default_response = SUCCESS_STATUS.include?(transaction_current_status || request_status || transaction_status)
+
+        case action
+        when 'refund'
+          successful_current_status = transaction_current_status&.upcase == 'CANCELLED' && request_status&.downcase == 'success'
+          successful_current_status || default_response
```

Comment:
If `current_status = "APPROVED"` and `request_status = "failure"` it looks to me like this would be marked as successful because on line 297 success status would inclued "APPROVED" but my understanding is for a refund this combo should be failed. Is that right?

Reply:
Ah that's a good point, let me add some validations. I'm don't think that response should be returned but if it's is it's good to add some validations to prevent some false positives